### DISCUSSION TITLE
zh/getting-started

### DISCUSSION THREAD

**Discussion by giscus** [tzfun/vue-web-terminal#119] [Announcements]
# zh/getting-started

一个轻量、功能强大的命令式网页仿真终端插件

http://www.beifengtz.com/zh/getting-started.html

**Comment by demmon2** [comment]
多窗口示例有没有，如何维持多个终端的管理，如激活，不激活等

**Reply by tzfun** [reply]
> 多窗口示例有没有，如何维持多个终端的管理，如激活，不激活等

可以参考文档示例里面的这段代码：https://github.com/tzfun/vue-web-terminal/blob/doc/docs/.vuepress/components/TerminalLocalDemo.vue#L256-L266

效果：https://tzfun.github.io/vue-web-terminal/demo.html

**Reply by demmon2** [reply]
感谢大佬，上一个问题结合自定义数据模型，加v-show解决了，但是现在有新需求，有没有每一行输出的插槽，因为我想自定义每一行的样式，输出的内容的等，然后折叠命令的时候，能否控制折叠所有或者展开所有，然后能够获取到光标的位置信息等，clear清除命令，我有时候不想清除缓存，只想滚动到光标的位置， 综上三个需求 ： 1.能否自定义每一行的内容样式（最好提供一个插槽） 2.折叠命令能否全局的控制全部折叠-全部不折叠，部分折叠（最好提供api进行控制） 3.清除命令能否只是不清楚缓存，只是滚动到光标位置，

**Reply by tzfun** [reply]
1. 自定义每一行的内容样式是指命令行还是输出行？命令行目前没有插槽的，输出行有
2. 全部折叠及展开的API是有的：[switchAllFoldState](https://tzfun.github.io/vue-web-terminal/api.html#switchallfoldstate)
3. clear命令的本意就是清除记录或者历史命令缓存，你的“滚动道光标位置”的需求似乎与它不符，提供一个跳转到窗口底部的API是不是可以解决你的问题？

**Comment by demmon2** [comment]
1. 命令行，我想在每一个命令行的前面加上执行时间，并且可控制显示或者不显示
2. 收到
3. 我的表达有误，就是想要跳转窗口底部

**Reply by demmon2** [reply]
折叠按钮的样式可以自定义吗

**Reply by tzfun** [reply]
可以的，我加一个优化单下个版本更新一下。

**Reply by tzfun** [reply]
> 折叠按钮的样式可以自定义吗

你想自定义到什么程度？

**Comment by demmon2** [comment]
就是能自定义展开与折叠的图标,&nbsp; &nbsp;展开的线条能否控制显示与不显示， 还有能否自定义线条的样式




------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2024年11月22日(星期五) 上午10:32
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [tzfun/vue-web-terminal] zh/getting-started (Discussion #119)














  
折叠按钮的样式可以自定义吗
  
你想自定义到什么程度？

**Reply by tzfun** [reply]
这些自定义有点细致，要单独控制得设置多个Slot和Attribute，我不太想因为一个显示细节设置太多Attribute和Slot，这会让插件变得臃肿，不过提供一个包含整个展开图标及线条的Slot是可以的，这样需要用户自己定义所有样式和点击事件

**Comment by demmon2** [comment]
可以的，感谢大佬，只要能提供改的入口就行




------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2024年11月22日(星期五) 上午10:46
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [tzfun/vue-web-terminal] zh/getting-started (Discussion #119)














 
这些自定义有点细致，要单独控制得设置多个Slot和Attribute，我不太想因为一个显示细节设置太多Attribute和Slot，这会让插件变得臃肿，不过提供一个包含整个展开图标及线条的Slot是可以的，这样需要用户自己定义所有样式和点击事件

**Comment by demmon2** [comment]
大佬在打扰一下呢，有没有获取光标信息的api



------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2024年11月22日(星期五) 上午10:01
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [tzfun/vue-web-terminal] zh/getting-started (Discussion #119)














  
自定义每一行的内容样式是指命令行还是输出行？命令行目前没有插槽的，输出行有
 
全部折叠及展开的API是有的：switchallfoldstate
 
clear命令的本意就是清除记录或者历史命令缓存，你的“滚动道光标位置”的需求似乎与它不符，提供一个跳转到窗口底部的API是不是可以解决你的问题？

**Reply by tzfun** [reply]
这个API没有提供，但在用户输入时搜索回函数中调会提供： [input-tips-search-handler](https://tzfun.github.io/vue-web-terminal/attributes.html#input-tips-search-handler)

**Comment by demmon2** [comment]
好的，感谢



------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2024年11月22日(星期五) 下午3:35
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [tzfun/vue-web-terminal] zh/getting-started (Discussion #119)














 
这个API没有提供，但在用户输入时搜索回函数中调会提供： [input-tips-search-handler
